includes/events.js

```diff
@@ -1,63 +1,54 @@
+            request_query: "ARRAY of STRUCTs, each with a key and a value. Contains any query parameters that were sent to the application as part of the web request that was this event or caused this event.",
+            response_content_type: "Content type of any data that was returned to the browser following the web request that either was this event or caused this event. For example, 'text/html; charset=utf-8'. Image views, for example, may have a non-text/html content type.",
+            response_status: "HTTP response code returned by the application in response to the web request that either was this event or caused this event. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Status.",
+            data: {
+                description: "ARRAY of STRUCTs, each with a key and a value. Contains a set of data points appropriate to the event_type of this event. For example, if this event was an entity create, update, delete or import event, data will contain the values of each field in the database after this event took place - according to the settings in the analytics.yml configured for this instance of dfe-analytics. Value be anonymised as a one way hash, depending on configuration settings.",
+                columns: {
+                    key: "Name of the field in the entity_table_name table in the database after it was created or updated, or just before it was imported or destroyed.",
+                    value: "Contents of the field in the database after it was created or updated, or just before it was imported or destroyed."
+                }
+            },
+            entity_table_name: "If event_type was an entity create, update, delete or import event, the name of the table in the database that this entity is stored in. NULL otherwise.",
+            anonymised_user_agent_and_ip: "One way hash of a combination of the user's IP address and user agent. Can be used to identify the user anonymously, even when user_id is not set. Cannot be used to identify the user over a time period of longer than about a month, because of IP address changes and browser updates.",
+            device_category: "The category of device used to cause this event - desktop, mobile, bot or unknown.",
+            browser_name: "The name of the browser used to cause this event.",
+            browser_version: "The version of the browser used to cause this event.",
+            operating_system_name: "The name of the operating system used to cause this event.",
+            operating_system_vendor: "The vendor of the operating system used to cause this event.",
+            operating_system_version: "The version of the operating system used to cause this event."
+        }
+    }).query(ctx => `WITH
+  earliest_web_request_event_for_request AS (
   SELECT
```

Comment:
Completely unnecessary CTE that I've replaced with an analytics ROW_NUMBER() function in line 70 in the new file